packages/core/src/checkout-buttons/create-checkout-headless-button-registry-v2.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    CheckoutButtonStrategy,
+    CheckoutButtonStrategyFactory,
+    CheckoutButtonStrategyResolveId,
+    isResolvableModule,
+    PaymentIntegrationService,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { ResolveIdRegistry } from '../common/registry';
+import * as defaultCheckoutHeadlessButtonStrategyFactories from '../generated/checkout-headless-button-strategies';
+
+export interface CheckoutButtonStrategyFactories {
+    [key: string]: CheckoutButtonStrategyFactory<CheckoutButtonStrategy>;
+}
+
+export default function createCheckoutHeadlessButtonStrategyRegistry(
```

Comment:
This function looks like a duplicate of `createCheckoutButtonStrategyRegistry`, except that the default value for `checkoutButtonHeadlessStrategyFactories` parameter is different. Instead of duplicating, could we just call `createCheckoutButtonStrategyRegistry` and pass it with `defaultCheckoutHeadlessButtonStrategyFactories` instead? It should work for both headless and non-headless because both implement the same interface.

Reply:
@davidchin yes, it should work. But based on [this](createCheckoutButtonRegistryV2) comment we need to create separate interface for Headless and at the end eventually we need this function